Add CDPx pipeline

lilian-wang opened the pull request:
Add scripts required for getting WebRTC code base, patching it, and building it for UWP

lilian-wang committed:
Add CDPx pipeline

LoadLibrary commented on .pipelines/build.cmd:
```diff
@@ -0,0 +1,28 @@
+
+set PYTHONPATH=
+set PYTHONHOME=
+set DEPOT_TOOLS_WIN_TOOLCHAIN=0
```
Please add "set GYP_MSVS_VERSION=2019" as well.

LoadLibrary commented on .pipelines/build.cmd:
```diff
@@ -0,0 +1,28 @@
+
+echo.
+echo Viewing the full list of editable build arguments...
+call gn args --list out\msvc\x64\Release
```
Why do we need this?

LoadLibrary commented on .pipelines/build.cmd:
```diff
@@ -0,0 +1,28 @@
+echo.
+echo Copying contents...
+cd /D "%~dp0"
+xcopy /e /i c:\webrtc\src\out ..\output
```
Please add a new line in the end

LoadLibrary commented on .pipelines/restore-acquiring_code.cmd:
```diff
@@ -0,0 +1,89 @@
+mkdir c:\Downloads
+cd c:\Downloads
+curl -o vs_enterprise.exe -L "https://aka.ms/vs/16/release/vs_enterprise.exe"
+call C:\Downloads\vs_enterprise.exe modify --installPath "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise" --add Microsoft.VisualStudio.Component.VC.ATLMFC;includeRecommended --quiet --wait
```
I think we need more workloads than just ATL.
Please add:
--add Microsoft.VisualStudio.Workload.NativeDesktop ^
--add Microsoft.VisualStudio.Workload.Universal

lilian-wang replied:
These workloads are installed in the base image.

LoadLibrary replied:
ok

LoadLibrary commented on .pipelines/restore-acquiring_code.cmd:
```diff
@@ -0,0 +1,89 @@
+
+echo.
+echo Installing Windows Debugging Tools...
+curl -o winsdksetup.exe -L "https://go.microsoft.com/fwlink/p/?linkid=2083338&clcid=0x409"
```
Can you add a comment saying which SDK version will this install?

LoadLibrary commented on .pipelines/restore-acquiring_code.cmd:
```diff
@@ -0,0 +1,89 @@
+echo.
+echo Installing Windows Debugging Tools...
+curl -o winsdksetup.exe -L "https://go.microsoft.com/fwlink/p/?linkid=2083338&clcid=0x409"
+winsdksetup.exe /features OptionId.WindowsDesktopDebuggers /quiet
```
What happens if machine have more than one SDK installed? The UWP workload from VS installer might install a SDK other that 18362 in near future.

lilian-wang replied:
This installs the Debugging Tools alone, without the SDK.

LoadLibrary replied:
For which SDK version?

lilian-wang replied:
Added comment in the previous line that this installs the Debugging Tools alone for Windows 10 SDK, version 1903.

LoadLibrary commented on .pipelines/restore-acquiring_code.cmd:
```diff
@@ -0,0 +1,89 @@
+
+REM Getting depot_tools
+echo.
+echo Cloneing the depot_tools repo in the root of the c:drive...
```
Cloneing -> Cloning

LoadLibrary commented on .pipelines/restore-acquiring_code.cmd:
```diff
@@ -0,0 +1,89 @@
+
+echo.
+echo Downloading gclient...
+echo N | gclient
```
Can you add a comment explaining why you're passing "N" to gclient?
We remember the reason now (because the most recent version of depot_tools dropped compatibility with python 2.7 that is currently required for building for Windows by some components), we might forget in the future.

LoadLibrary commented on .pipelines/restore-acquiring_code.cmd:
```diff
@@ -0,0 +1,89 @@
+REM Installing Python 2.7
+echo.
+echo Making sure that any python program executed in this session comes from the depot_tools...
+where python
```
What does the script do with the line?

LoadLibrary commented on .pipelines/restore-acquiring_code.cmd:
```diff
@@ -0,0 +1,89 @@
+
+echo.
+echo Installing pip...
+call python -m pip install --upgrade pip
```
Can the script check if the right python is being called, log an error message and stop execution?

lilian-wang replied:
The previous command "where python" confirmed the right python location.

LoadLibrary replied:
Thanks for adding that check.

LoadLibrary reviewed (changes requested):
It would be good to have some error handling. Something like:

the_command_to_execute.exe
if errorlevel 1 (
   echo Last command failed with erro code: %errorlevel%
   exit /b %errorlevel%
)
It would save us time in the future when some dependency changes.

lilian-wang committed:
Modify scripts

- Add error handling
- Set GYP_MSVS_VERSION in build script
- Add new line in the end
- Add comment on SDK version installed
